src/Glpi/Controller/GenericFormController.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Controller;
+
+use CommonDBTM;
+use Html;
+use Glpi\Event;
+use Glpi\Exception\Http\AccessDeniedHttpException;
+use Glpi\Exception\Http\BadRequestHttpException;
+use Glpi\Exception\Http\NotFoundHttpException;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Attribute\Route;
+
+class GenericFormController extends AbstractController
+{
+    private const SUPPORTED_ACTIONS = [
+        'add',
+        'delete',
+        'restore',
+        'purge',
+        'update',
+        'unglobalize',
+    ];
```

Comment:
Do you think it would make sense to create a public enum instead ?
This way it could be reused in forms inputs and avoid repeating these hardcoded string everywhere.

Reply:
We could, but it would make sense to do it if we replace most of the hardcoded value in our whole codebase. I guess it would be preferable to do it later, this PR is already big enough.

Reply:
There's also an issue for "custom actions" that we might want to create.

There's dozens of other form files that are not migrated yet and we don't know yet how many of them have a specific action that might be necessary to handle, whether a custom controller must be created, or a custom implementation somewhere else, etc., and an Enum prevents this, while a simple string is more flexible.